src/main/java/bern/functionalities/Ui.java

```diff
@@ -0,0 +1,276 @@
+import bern.exception.BernException;
+import bern.exception.EmptyDescriptionException;
+import bern.exception.IndexException;
+import bern.exception.InvalidCommandException;
+import bern.model.Deadline;
+import bern.model.Event;
+import bern.model.Task;
+import bern.model.ToDo;
+
+/**
+ * A class to encapsulates all methods UI related.
+ */
+public class Ui {
+
+    private Parser parser = new Parser();
+    private Storage storage = new Storage();
+
+    /**
+     * A method to get the reply if input starts with deadline.
+     *
+     * @param input The given command.
+     * @param arListTask The initial ArrayList of Tasks.
+     * @return The reply by the bot.
+     * @throws BernException If input is invalid.
+     */
+    public String ifDeadline(String input, ArrayList<Task> arListTask) throws BernException {
+        if (input.length() == 8 || (input.length() == 9 && input.substring(8, 9).equals(" "))) {
+            throw new EmptyDescriptionException("deadline");
+        }
+        // TODO Haven't handled if there is no /by
+        String task = input.substring(9, input.indexOf('/') - 1);
+        String by = input.substring(input.indexOf('/') + 4);
+        new TaskList().addTask(new Deadline(task, by), arListTask);
+        return "Got it. I've added this task:\n"
+                + arListTask.get(arListTask.size() - 1).toString() + "\n"
+                + "Now you have " + String.valueOf(arListTask.size())
+                + (arListTask.size() == 1 ? " task in the list" : " tasks in the list");
+    }
+
+    /**
+     * A method to get the reply if input starts with event.
+     *
+     * @param input The given command.
+     * @param arListTask The initial ArrayList of Tasks.
+     * @return The reply by the bot.
+     * @throws BernException If input is invalid.
+     */
+    public String ifEvent(String input, ArrayList<Task> arListTask) throws BernException {
+        if (input.length() == 5 || (input.length() == 6 && input.substring(5, 6).equals(" "))) {
+            throw new EmptyDescriptionException("event");
+        }
+        // TODO Haven't handled if there is no /at
+        String task = input.substring(6, input.indexOf('/') - 1);
+        String at = input.substring(input.indexOf('/') + 4);
+        new TaskList().addTask(new Event(task, at), arListTask);
+        return "Got it. I've added this task:\n"
+                + arListTask.get(arListTask.size() - 1).toString() + "\n"
+                + "Now you have " + String.valueOf(arListTask.size())
+                + (arListTask.size() == 1 ? " task in the list" : " tasks in the list");
+    }
+
+    /**
+     * A method to get the reply if input starts with todo.
+     *
+     * @param input The given command.
+     * @param arListTask The initial ArrayList of Tasks.
+     * @return The reply by the bot.
+     * @throws BernException If input is invalid.
+     */
+    public String ifToDo(String input, ArrayList<Task> arListTask) throws BernException {
```

Comment:
The function name was a little misleading to me. Maybe consider a name that describes its purpose? (like `replyToDo()`)

I think I prefer this because its purpose of checking format is secondary as compared to its purpose of generating a reply.

Reply:
Can consider applying this to the other reply methods too!